dbms/src/IO/BrotliWriteBuffer.cpp

```diff
@@ -0,0 +1,119 @@
+#include <Common/config.h>
+#if USE_BROTLI
+
+#include <IO/BrotliWriteBuffer.h>
+#include <brotli/encode.h>
+
+namespace DB
+{
+
+class BrotliWriteBuffer::BrotliStateWrapper
+{
+public:
+    BrotliStateWrapper()
+    : state(BrotliEncoderCreateInstance(nullptr, nullptr, nullptr))
+    {
+    }
+
+    ~BrotliStateWrapper()
+    {
+        BrotliEncoderDestroyInstance(state);
+    }
+
+public:
+    BrotliEncoderState * state;
+};
+
+BrotliWriteBuffer::BrotliWriteBuffer(WriteBuffer & out_, int compression_level, size_t buf_size, char * existing_memory, size_t alignment)
+        : BufferWithOwnMemory<WriteBuffer>(buf_size, existing_memory, alignment)
+        , brotli(new BrotliStateWrapper())
+        , in_available(0)
+        , in_data(nullptr)
+        , out_capacity(0)
+        , out_data(nullptr)
+        , out(out_)
+{
+    BrotliEncoderSetParameter(brotli->state, BROTLI_PARAM_QUALITY, static_cast<uint32_t>(compression_level));
+    BrotliEncoderSetParameter(brotli->state, BROTLI_PARAM_LGWIN, 24);
```

Comment:
Is it the recommended value from the library docs/examples? Comment needed.

Reply:
added reference to brotli source code